homeassistant/components/ruuvi_gateway/bluetooth.py

```diff
@@ -0,0 +1,102 @@
+"""Bluetooth support for Ruuvi Gateway."""
+from __future__ import annotations
+
+from collections.abc import Callable
+import datetime
+import logging
+
+from home_assistant_bluetooth import BluetoothServiceInfoBleak
+
+from homeassistant.components.bluetooth import (
+    BaseHaRemoteScanner,
+    async_get_advertisement_callback,
+    async_register_scanner,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+
+from .coordinator import RuuviGatewayUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RuuviGatewayScanner(BaseHaRemoteScanner):
+    """Scanner for Ruuvi Gateway."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        scanner_id: str,
+        name: str,
+        new_info_callback: Callable[[BluetoothServiceInfoBleak], None],
+        *,
+        coordinator: RuuviGatewayUpdateCoordinator,
+    ) -> None:
+        """Initialize the scanner, using the given update coordinator as data source."""
+        super().__init__(
+            hass,
+            scanner_id,
+            name,
+            new_info_callback,
+            connector=None,
+            connectable=False,
+        )
+        self.coordinator = coordinator
+
+    @callback
+    def _async_handle_new_data(self) -> None:
+        now = datetime.datetime.now()
+        for tag_data in self.coordinator.data:
+            if now - tag_data.datetime > datetime.timedelta(minutes=10):
```

Comment:
Can we use the stale advertisement constants from the Bluetooth integration here?